File: src/Services/Notification/Notification.API/Endpoints/UserNotification/CreateUserNotification.cs
using BuildingBlock.Application.Abstractions.Common;
using BuildingBlock.Infrastructure.Authorization;

using Notification.Application.Features.UserNotifications.Commands.CreateUserNotification;

namespace Notification.API.Endpoints.UserNotification;

/// <summary>Admin-only - no automatic rule/campaign trigger is wired up yet to generate these.</summary>
public sealed class CreateUserNotification : ICarterModule
{
    public void AddRoutes(IEndpointRouteBuilder app)
    {
        app.MapPost("/user-notifications", HandleAsync)
            .RequireAuthorization(AuthorizationPolicies.RequireAdmin)
            .WithName("CreateUserNotification")
            .WithDisplayName("Create User Notification API")
            .WithDescription("Creates a Notification Center entry for a user. Admin only - no automatic rule/campaign trigger is wired up yet.")
            .Produces<ApiResponse<CreateUserNotificationResponse>>(StatusCodes.Status200OK);
    }

    private static async Task<IResult> HandleAsync(
        [FromBody] CreateUserNotificationCommand command,
        [FromServices] ISender sender,
        CancellationToken ct = default)
    {
        var response = await sender.Send(command, ct);
        return Results.Ok(ApiResponse<CreateUserNotificationResponse>.Ok(response));
    }
}

File: src/services/notification/create-user-notification.ts
import { apiClient } from '@/shared/lib/api/client';
import { BASE_PATH } from './_base';
import type { NotificationPriority } from './types/enums';

export interface CreateUserNotificationCommand {
  userId: string;
  category?: string;
  type?: string;
  title: string;
  body: string;
  priority: NotificationPriority;
  metadataJson?: string;
  expiredAt?: string;
  campaignId?: string;
}

export interface CreateUserNotificationResponse {
  id: string;
}

/**
 * POST /user-notifications — admin only. Creates a Notification Center entry
 * for a user; no automatic rule/campaign trigger is wired up yet.
 */
export function createUserNotification(
  command: CreateUserNotificationCommand,
): Promise<CreateUserNotificationResponse> {
  return apiClient.post(`${BASE_PATH}/user-notifications`, command).then((res) => res.data);
}
